add talk listing to home page

wgao19 opened the pull request:
Implements #41 

cc @JennieJi  @huijing

nutstick commented on src/pages/index.jsx:
```diff
@@ -168,6 +193,23 @@ export const pageQuery = graphql`
       }
     }
+    content: github {
+      repository(owner: "react-knowledgeable", name: "talks") {
```
We can also merge this `query` with `talks query` below as well

nutstick replied:
But leave it like this also fine. Maybe it's more clean. XD

wgao19 replied:
I'm wondering, how does the merge work? The first content query is meant to be grouped by labels.

nutstick replied:
Sorry, I mean you can do.
```gql
{
  github {
    repository(owner: "react-knowledgeable", name: "talks") {
      content: labels {
         ....
      }
      talks: issues {
         ....
      }
    }
  }
}
```
Which it should request data from github once and got 2 data instead of request 2 times, I guess.

wgao19 replied:
updated, thanks for the suggestion :)

thchia commented:
Sorry I anyhow press resolve conversation on the query discussion. But I think you could have just had two fields in the one query.

wgao19 commented:
Thanks for the work and suggestions! Here are some of my thoughts:

- is there any way to make the boxes take their own content height?

- on the widest screen, the part on the RHS feels a bit too much information stacked into one small region.. maybe can adjust a little bit, move something to other places, etc:
![image](https://user-images.githubusercontent.com/2055384/67625430-dd0aaa00-f870-11e9-9878-b7a6fb18e580.png)

- on narrower screens, shall we put "on the next RK" before "past talks"?
![image](https://user-images.githubusercontent.com/2055384/67625439-f4499780-f870-11e9-8dd4-7b7af1ad8911.png)

huijing commented:
Looks overly-long right now, but will scale well once you have more talks in the past talks section.
![Screenshot_2019-10-27 React Knowledgeable · Fun and friendly podium to share what we learn about React ](https://user-images.githubusercontent.com/1461498/67634067-b64e7280-f874-11e9-9e28-f9c3d1626fdc.png)

Changing the visual order will affect keyboard navigation, unless u are able to modify the markup at a different viewport size.

wgao19 commented:
it looks nicer than the last version 😃 

peaking from afar, what do u think of moving stories to beneath the main section to make lengths match? 
on the flip side, this main section is going to grow longer, so im fine w the current layout too

gonna take whatever ur call on this is, and thank you muchly for working on this again 🐳💜

wgao19 commented:
i really love how the main section flows the boxes btw, ur so great at this 🦔🦔🦔🦔

huijing commented:
i think after 2 more meetups, the past talks content will become long-ish enough to even things out (so it will look nicer in 2 months?)
but is @JennieJi ideas v good :)
also u r director...haha...i'm just the guy who moves things around ¯\_(ツ)_/¯

wgao19 commented:
cool, let's merge this :) thank you all!

wgao19 commented:
Erm, so, girls, sorry that I then made a lot more changes than expected. Mainly because our changes to home page layout breaks other pages. And I've refactored and simplified our structure a little bit. Then some further changes incurred.

In summary:
- refactored the whole structure to have a default layout for all pages
- the default layout is a simple header - main + aside - footer
- added talk content by parsing GH issue body (my bad, I should have foreseen and done this before having you working on this)
- remove unused css and other minor refactors

I will merge this for now. A lot of my implementation is now pretty crude, but hopefully it's conceptually simpler for easier maintenance in the future. And we can pick up from there:
- bring back "submit talk" button on home page
- the home page now looks slightly clumpsy because there is a lot of content now, I think we can find ways to optimize this
- footer is very delicate now (i.e., in "story" page, if you add a border top to match the border bottom on header, you can see the width is broken)

wgao19 commented:
addresses #14 on talk parsing as wel
